disperser/meterer/types.go

```diff
@@ -0,0 +1,243 @@
+package meterer
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/signer/core/apitypes"
+)
+
+/* SUBJECT TO BIG MODIFICATIONS */
+
+// BlobQuorumParam represents the parameters for a specific quorum
+type BlobQuorumParam struct {
+	QuorumID           core.QuorumID
+	AdversaryThreshold uint32
+	QuorumThreshold    uint32
+}
+
+// BlobHeader represents the header information for a blob
+type BlobHeader struct {
+	// Existing fields
+	Commitment       core.G1Point
+	DataLength       uint32
+	BlobQuorumParams []BlobQuorumParam
+
+	// New fields
+	Version   uint32
+	AccountID string
+	Nonce     uint32 // use nonce to prevent duplicate payments in the same reservation window
+	BinIndex  uint64
```

Comment:
It seems like `uint32` would be sufficient for this?

Reply:
updated to be more memory efficient; I'm assuming we don't need to worry about an index in the unit of seconds (previously global bin index is in seconds so 2^32 in unix let this go up to Jan 19, 2038, but with that being removed, uint32 is fine for reservation bin indices

I also updated start & end epochs to use `uint32`, as well as `PricePerChargeable` (2^64 gwei ~= 18M Eth; uint32 => ~4ETH), `MinChargeableSize` (in bytes, this is ~4GB/s), and `ReservationWindow` (136years but uint16 is only 18hours if we are counting in seconds; we could instead use hour as window unit and allow for 7years). 

Left `GlobalBytesPerSecond` with `uint64` for the aspiration of supporting rate-limit greater than 4GB/s network-wide ✨